### ISSUE TITLE
bob sync alwas returns error

### ISSUE THREAD

**Issue by indika-dev** [MordechaiHadad/bob#109]
Hi, 

everytime I run `bob sync`, I'll get the following error:

> $ bob sync
Mar 13 10:44:55.490  INFO Starting sync process
Mar 13 10:44:55.491  INFO Using version v0.8.3
 set in /home/xyz/.config/bob/nvim.version
Mar 13 10:44:55.491 ERROR Error: Please provide a proper version string

My config file in ~/.config/bob:

> $ cat config.json 
{
  "enable_nightly_info": true,
  "downloads_location": "/home/xyz/.local/share/bob",
  "installation_location": "/home/xyz/.local/share/bob/nvim-bin",
  "version_sync_file_location": "/home/xyz/.config/bob/nvim.version",
  "rollback_limit": 3
}

My nvim.version in  ~/.config/bob:
> v0.8.3

Bob was installed via cargo:
`cargo 1.67.1 (8ecd4f20a 2023-01-10)`
`rustc 1.67.1 (d5a82bbd2 2023-02-07)`
OS is Ubuntu 22.04.2

I don't understand, why I get this error. I checked the version string online against the regular expression from bob, I tried "stable" and "nightly" as version strings, but I always get the described error.
Any clues, what's going wrong?

**Comment by MordechaiHadad** [comment]
@indika-dev are those paths in your config file really contain `/home/xyz` or did you change xyz to your username(as you should)?

**Comment by indika-dev** [comment]
I wrote `xyz` instead of my real username only for this bug report. The real config files have always my real username on this laptop.

**Comment by MordechaiHadad** [comment]
fair enough

**Comment by MordechaiHadad** [comment]
Can you run `bob sync` when downloading bob PR specifically this https://github.com/MordechaiHadad/bob/suites/11523839352/artifacts/595999052 
and if it still happens, I am not sure it will work but currently I am unable to reproduce your issue on my end.

**Comment by indika-dev** [comment]
Unfortunately, that does not work either. I also have absolutley no idea, why sync is not working. bob reads the correct version information of the file, I checked the regex against content of the version file and I tested stable/nightly as versions, which are compared directly and it's still not working.

**Comment by MordechaiHadad** [comment]
I see, thank you.
I will dig more into this!

**Comment by MordechaiHadad** [comment]
@indika-dev I think I found what is causing this error and it makes no sense, try changing the location of `nvim.version` to be somewhere else besides `~/.config/nvim/` this is how it breaks for me

**Comment by MordechaiHadad** [comment]
@indika-dev what happens if you run `bob erase` `bob use stable` `bob erase` and `bob sync`

**Comment by indika-dev** [comment]
your command sequence works!! I used the version, which you [mentioned before](https://github.com/MordechaiHadad/bob/issues/109#issuecomment-1466154367). 

>$ bob erase  
Mar 14 11:26:07.321  INFO Successfully removed neovim's installation folder  
Mar 14 11:26:07.369  INFO Successfully removed neovim downloads folder  
$ bob use stable
Mar 14 11:26:15.706  INFO Fetching latest version
Downloaded version v0.8.3 to /home/xyz/.local/share/bob/v0.8.3.tar.gz
  [00:00:00] [██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 9.98MiB/9.98MiB (43.35MiB/s, 0s)
Finished expanding to /home/xyz/.local/share/bob/v0.8.3
  [00:00:00] [█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 1692/1692
Mar 14 11:26:17.091  INFO Written version to /home/xyz/.config/bob/nvim.version
Mar 14 11:26:17.091  INFO You can now use v0.8.3!
$ bob erase
Mar 14 11:26:24.128  INFO Successfully removed neovim's installation folder
Mar 14 11:26:24.173  INFO Successfully removed neovim downloads folder
$ bob sync
Mar 14 11:26:29.866  INFO Starting sync process
Mar 14 11:26:29.866  INFO Using version v0.8.3 set in /home/xyz/.config/bob/nvim.version
Downloaded version v0.8.3 to /home/xyz/.local/share/bob/v0.8.3.tar.gz
  [00:00:00] [██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 9.98MiB/9.98MiB (19.46MiB/s, 0s)
Finished expanding to /home/xyz/.local/share/bob/v0.8.3
  [00:00:00] [█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 1692/1692
Mar 14 11:26:31.068  INFO You can now use v0.8.3!
$ 

It looks like, that it has something to do with the editor:
- after your command sequence
>~ /.config/bob$ cat nvim.version 
v0.8.3~/.config/bob$ ls
- I edited it with neovim and readded the same version information
>~/.config/bob$ cat nvim.version 
v0.8.3  
~/.config/bob$ bob sync

It looks like, that at least neovim adds something like a linebreak, even if I didn't enter return.

**Comment by MordechaiHadad** [comment]
Yeah it seems like when bob adds the content to the file it doesn't add [LF character](https://developer.mozilla.org/en-US/docs/Glossary/CRLF) , I can probably remove this character after reading the file

**Comment by MordechaiHadad** [comment]
@indika-dev I think I have fixed it, try using `bob sync` when u manually edited `nvim.version` using https://github.com/MordechaiHadad/bob/actions/runs/4415137849

**Comment by MordechaiHadad** [comment]
> for reference:
> 
>     * that is nvim.version after command sequence(output from xxd):
>       `00000000: 7630 2e38 2e33                           v0.8.3`
> 
>     * that is nvim.version after editing with neovim(output from xxd):
>       `00000000: 7630 2e38 2e33 0a                        v0.8.3.`

Yeah I used hexyl to check this

**Comment by MordechaiHadad** [comment]
last commit should deal with this trailing character

**Comment by indika-dev** [comment]
@MordechaiHadad build failed

**Comment by MordechaiHadad** [comment]
> @MordechaiHadad build failed

This one then https://github.com/MordechaiHadad/bob/actions/runs/4415175805

**Comment by indika-dev** [comment]
I can confirm, that it is working! 
> :~/.config/bob$ bob sync
Mar 14 12:57:40.331  INFO Starting sync process
Mar 14 12:57:40.331  INFO Using version v0.8.3
 set in /home/xyz/.config/bob/nvim.version
Mar 14 12:57:40.332  INFO v0.8.3 is already installed and used!

**Comment by MordechaiHadad** [comment]
@indika-dev thank you for testing!
